internal/controller/genericprovider_controller.go

```
@@ -282,6 +290,49 @@ func addConfigSecretToHash(ctx context.Context, k8sClient client.Client, hash ha
 	return nil
 }
 
+func addConfigMapToHash(ctx context.Context, k8sClient client.Client, hash hash.Hash, provider genericprovider.GenericProvider) error {
+	spec := provider.GetSpec()
+	if spec.FetchConfig != nil && spec.FetchConfig.Selector != nil {
+		// List ConfigMaps that match the provider's selector
+		selector, err := metav1.LabelSelectorAsSelector(spec.FetchConfig.Selector)
+		if err != nil {
+			return err
+		}
+
+		configMapList := &corev1.ConfigMapList{}
+		listOpts := []client.ListOption{
+			client.MatchingLabelsSelector{Selector: selector},
+			client.InNamespace(provider.GetNamespace()),
+		}
+
+		if err := k8sClient.List(ctx, configMapList, listOpts...); err != nil {
+			return err
+		}
+
+		// Sort the ConfigMaps by name for consistent hashing
+		configMaps := configMapList.Items
+		for i := 0; i < len(configMaps)-1; i++ {
+			for j := i + 1; j < len(configMaps); j++ {
+				if configMaps[i].Name > configMaps[j].Name {
+					configMaps[i], configMaps[j] = configMaps[j], configMaps[i]
+				}
+			}
+		}
```

Could you please return an error in case of more than 1 matching config map? This should not be allowed.

---

Thanks for your review @Danil-Grigorev . I have added error handling in the latest commit.